walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,108 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    // Only works with sorted, non-overlapping time ranges
+    // checkTimes prepares data
+    private Collection<TimeRange> invert(Collection<TimeRange> booked_times, long requestDuration){
+        Collection<TimeRange> possible_times = new ArrayList<TimeRange>();
+        int firstTime = 0;
+        for(TimeRange tr : booked_times){
+            TimeRange free = TimeRange.fromStartEnd(firstTime, tr.start(), false);
+
+            // Ensures a meeting ending before the endpoint of a previous meeting cannot allow
+            // an invalid meeting to be scheduled
+            if(tr.start() + tr.duration() > firstTime)
```

Comment:
Isn't `tr.start()+tr.duration()` the same as `tr.end()`?

When would `firstTime` not be less than `tr.end()` since it starts at 0 and only will ever be set to the last time range's end value? I think in your comment you said that this method assumes that time ranges are sorted and non-overlapping, so I'm not sure if you need this conditional.